cpp/src/strings/regex/regex.cuh

```diff
@@ -204,34 +182,65 @@ class reprog_device {
    */
   template <int stack_size>
   __device__ inline match_result extract(cudf::size_type idx,
-                                         string_view const& d_str,
+                                         string_view const d_str,
                                          cudf::size_type begin,
                                          cudf::size_type end,
-                                         cudf::size_type group_id);
+                                         cudf::size_type const group_id) const;
 
  private:
-  int32_t _startinst_id, _num_capturing_groups;
-  int32_t _insts_count, _starts_count, _classes_count;
-  const uint8_t* _codepoint_flags{};  // table of character types
-  reinst* _insts{};                   // array of regex instructions
-  int32_t* _startinst_ids{};          // array of start instruction ids
-  reclass_device* _classes{};         // array of regex classes
-  void* _relists_mem{};               // runtime relist memory for regexec
+  struct reljunk {
+    relist* list1;
+    relist* list2;
+    int32_t starttype{};
+    char32_t startchar{};
```

Comment:
Same here. Anything we can constify/restrict the pointers?

Reply:
The `list1` and `list2` pointers are modified by the `swaplist()` function. The `starttype` and the `startchar` are set in the ctor with some if-statement logic.